core/src/main/java/org/apache/iceberg/RESTTableScan.java

```diff
@@ -0,0 +1,258 @@
+      statsFields =
+          columnsToKeepStats().stream()
+              .map(columnId -> schema().findColumnName(columnId))
+              .collect(Collectors.toList());
+    }
+
+    PlanTableScanRequest.Builder planTableScanRequestBuilder =
+        new PlanTableScanRequest.Builder()
+            .withSelect(selectedColumns)
+            .withFilter(filter())
+            .withCaseSensitive(isCaseSensitive())
+            .withStatsFields(statsFields);
+
+    if (startSnapshotId != null && endSnapshotId != null) {
+      planTableScanRequestBuilder
+          .withStartSnapshotId(startSnapshotId)
+          .withEndSnapshotId(endSnapshotId)
+          .withUseSnapshotSchema(true);
+
+    } else if (snapshotId != null) {
+      boolean useSnapShotSchema = snapshotId != table.currentSnapshot().snapshotId();
+      planTableScanRequestBuilder
+          .withSnapshotId(snapshotId)
+          .withUseSnapshotSchema(useSnapShotSchema);
+
+    } else {
+      planTableScanRequestBuilder.withSnapshotId(table().currentSnapshot().snapshotId());
+    }
+
+    return planTableScan(planTableScanRequestBuilder.build());
+  }
+
+  private CloseableIterable<FileScanTask> planTableScan(PlanTableScanRequest planTableScanRequest) {
+    ParserContext context =
+        ParserContext.builder()
+            .add("specsById", table.specs())
+            .add("caseSensitive", context().caseSensitive())
+            .build();
+
+    PlanTableScanResponse response =
+        client.post(
+            resourcePaths.planTableScan(tableIdentifier),
+            planTableScanRequest,
+            PlanTableScanResponse.class,
+            headers.get(),
+            ErrorHandlers.defaultErrorHandler(),
+            stringStringMap -> {},
+            context);
+
+    PlanStatus planStatus = response.planStatus();
+    switch (planStatus) {
+      case COMPLETED:
+        return getScanTasksIterable(response.planTasks(), response.fileScanTasks());
+      case SUBMITTED:
+        return fetchPlanningResult(response.planId());
+      case FAILED:
+        throw new IllegalStateException(
+            "Received \"failed\" status from service when planning a table scan");
+      case CANCELLED:
+        throw new IllegalStateException(
+            "Received \"cancelled\" status from service when planning a table scan");
+      default:
+        throw new RuntimeException(
+            String.format("Invalid planStatus during planTableScan: %s", planStatus));
+    }
+  }
+
+  private CloseableIterable<FileScanTask> fetchPlanningResult(String planId) {
+    // TODO need to introduce a max wait time for this loop potentially
+    boolean planningFinished = false;
+    // we need to inject specById map here and also the caseSensitive
+    ParserContext context =
+        ParserContext.builder()
+            .add("specsById", table.specs())
+            .add("caseSensitive", context().caseSensitive())
+            .build();
+    while (!planningFinished) {
```

Comment:
Is this `planningFinished` state ever updated anywhere? It looks like in every possible branch we'll either exit due to a success or throw. The one exception is if it's repeatedly in submitted. But effectively as its written, this is just a while(true)...

Reply:
Discussed this offline but adding it here for the completion : 
Its supposed to be a while(True) loop essentially polling the server until the planning is completed because the `planFiles()` api has been called, this variable is essentially added to suppress the warning, per feedback i will change this to while(true) with a timeout and add suppress warning annotations.